elephas/ml_model.py

```diff
@@ -178,40 +187,40 @@ def _transform(self, df):
         """Private transform method of a Transformer. This serves as batch-prediction method for our purposes.
         """
         output_col = self.getOutputCol()
-        label_col = self.getLabelCol()
         new_schema = copy.deepcopy(df.schema)
-        new_schema.add(StructField(output_col, StringType(), True))
         rdd = df.rdd
-
         weights = self.weights
 
         def extract_features_and_predict(model_yaml: str,
                                          custom_objects: dict,
                                          features_col: str,
                                          model_type: ModelType,
+                                         predict_classes: bool,
                                          data):
             model = model_from_yaml(model_yaml, custom_objects)
             model.set_weights(weights.value)
-            predict_function = determine_predict_function(model, model_type)
+            predict_function = determine_predict_function(model, model_type, predict_classes)
             return predict_function(np.stack([from_vector(x[features_col]) for x in data]))
 
         predictions = rdd.mapPartitions(
             partial(extract_features_and_predict,
                     self.get_keras_model_config(),
                     self.get_custom_objects(),
                     self.getFeaturesCol(),
-                    self.model_type))
-        if self.model_type == ModelType.CLASSIFICATION:
-            predictions = predictions.map(lambda x: tuple(str(x)))
-        else:
+                    self.model_type,
+                    self.get_predict_classes()))
+        if (self.model_type == ModelType.CLASSIFICATION and self.get_predict_classes()) \
+                or self.model_type == ModelType.REGRESSION:
             predictions = predictions.map(lambda x: tuple([float(x)]))
+            output_col_field = StructField(output_col, DoubleType(), True)
+        else:
+            # we're doing classification and predicting class probabilities
+            predictions = predictions.map(lambda x: tuple([x.tolist()]))
+            output_col_field = StructField(output_col, ArrayType(DoubleType()), True)
```

Comment:
@OscarDPan  I think that's correct, although we're not doing argmax here - we only do argmax in the case we're using the keras functional API: https://github.com/maxpumperla/elephas/blob/master/elephas/utils/model_utils.py#L64. To your point, we may want to add a check there based on the loss.

Reply:
Ah yes that's the line I meant to comment on. :)

Reply:
My suggestion is instead of complicating Elephas's code by adding multiple checks, we can just implement the distributed predict() method and let end users decide how they want to transform the predicted probabilities.

I like Elephas for its simplicity and light weight and since there aren't many team members in this project we can focus more on the "distributed" features. Just my 2cents.

Reply:
Hi @danielenricocahall I found an edge case.
I am using a customized loss function call "tweedie loss", it's model type is supposed to be "ModelType.REGRESSION" but since it's not in the dictionary, in the end it goes to 
```        else:
            # we're doing classification and predicting class probabilities
            predictions = predictions.map(lambda x: tuple([x.tolist()]))
            output_col_field = StructField(output_col, ArrayType(DoubleType()), True)
```

The consequence is that all my existing code assuming the output should be a float gets error, since it is a list of one float. I was wondering if we can do some dynamic schema detection s.t. if len(output) == 1 then it's a DoubleType() otherwise it's ArrayType(DoubleType())?

Reply:
Ah! You can register the loss using:

`LossModelTypeMapper().register_loss('tweedie_loss', ModelType.REGRESSION)`

Reply:
Regarding your other suggestion of just making a distributed `predict`, I like it - however, that change may warrant a new major release, since any applications depending on `predict` and expect it to output a singular value would have to update their code (it's a breaking change).

Reply:
Cool! And thanks for the API tip!